### ISSUE TITLE
Support custon names for EnumClass

### ISSUE THREAD

**Issue by danielgomezrico** [google/built_value.dart#920]
I need to have an enum for countries, with values like `Mèxico` and just use my enum `Country.mexico` everywhere.

Currently I need to implement a mapper from the string to the enum, and I believe that this could be automatically done if the name of the enums could be customizable and so I can have custom `valueOf` autogenerated too.

# Current
`EnumClass.name` contains the name of the variable itself.

# Expected
I can override name for EnumClass with a custom string.

# Ideas
- Add a variable `name` in the `BuiltValueEnumConst`. that override this.

**Comment by davidmorgan** [comment]
This gets a bit complicated, you can already set `wireName`, but that doesn't affect `valueOf`.

Display feels like a different enough concept from what the enum is already doing, that it should maybe just be an additional method pair, e.g. `display` and `parse`, that you write by hand.

**Comment by danielgomezrico** [comment]
@davidmorgan why the `wireName` can not affect the `valueOf`?

**Comment by danielgomezrico** [comment]
@davidmorgan I'm asking because it feels totally natural to be able to overwrite this value, and also feels like a weird thing that serializer used `wiredName` while `valueOf` uses the variable name.

**Comment by danielgomezrico** [comment]
If I try to override that method and to it by myself `built_value` fails with:

```
[SEVERE] built_value_generator:built_value on lib/src/api/model/fee_measure_enum.dart:
Error in BuiltValueGenerator for /client/lib/src/api/model/fee_measure_enum.dart.
Please make the following changes to use EnumClass:

1. Add method: static FeeMeasureUnitEnum valueOf(String name) => _$valueOf(name)
```

So I cant do it even by myself.

**Comment by davidmorgan** [comment]
Right, you'll have to call it something different, e.g. `parse`. The implementation of `valueOf` is intentionally fixed.
